src/gui/torrentoptionsdialog.h

```diff
@@ -26,40 +26,67 @@
  * exception statement from your version.
  */
 
-#ifndef UPDOWNRATIODIALOG_H
-#define UPDOWNRATIODIALOG_H
+#pragma once
 
 #include <QDialog>
 
+namespace BitTorrent
+{
+    class TorrentHandle;
+    class InfoHash;
+}
+
 namespace Ui
 {
-    class UpDownRatioDialog;
+    class TorrentOptionsDialog;
 }
 
-class UpDownRatioDialog final : public QDialog
+class TorrentOptionsDialog final : public QDialog
 {
     Q_OBJECT
 
 public:
-    UpDownRatioDialog(bool useDefault, qreal initialValue, qreal maxValue,
-            int initialTimeValue, int maxTimeValue,
-            QWidget *parent = nullptr);
-    ~UpDownRatioDialog();
-
-    bool useDefault() const;
-    qreal ratio() const;
-    int seedingTime() const;
+    explicit TorrentOptionsDialog(QWidget *parent, const QVector<BitTorrent::TorrentHandle *> torrents);
+    ~TorrentOptionsDialog() override;
 
 public slots:
     void accept() override;
 
 private slots:
+    void updateUploadSpinValue(int val);
+    void updateDownloadSpinValue(int val);
+    void updateUploadSliderValue(int val);
+    void updateDownloadSliderValue(int val);
+
+    void handleUpSpeedLimitChanged();
+    void handleDownSpeedLimitChanged();
+    void handleShareLimitsChanged();
+    void handleDisableDHTChanged();
+    void handleDisablePEXChanged();
+    void handleDisableLSDChanged();
+
     void handleRatioTypeChanged();
-    void enableRatioSpin();
-    void enableTimeSpin();
 
 private:
-    Ui::UpDownRatioDialog *m_ui;
+    int getUploadSpeedLimit() const;
+    int getDownloadSpeedLimit() const;
+
+    bool useGlobalShareLimits() const;
+    qreal getRatio() const;
+    int getSeedingTime() const;
+
+    bool isDisableDHTChecked() const;
+    bool isDisablePEXChecked() const;
+    bool isDisableLSDChecked() const;
+
+    QVector<BitTorrent::InfoHash> m_torrentHashes;
+    bool m_upSpeedLimitChanged;
+    bool m_downSpeedLimitChanged;
+    bool m_shareLimitsChanged;
+    bool m_disableDHTChanged;
+    bool m_disablePEXChanged;
+    bool m_disableLSDChanged;
```

Comment:
I would give the default values here instead of in .cpp file.

Reply:
Done.